qiskit/qpy/interface.py

```diff
@@ -126,9 +129,23 @@ def dump(
             but not supported in all platforms. Please check that your target platform is supported
             by the symengine library before setting this option, as it will be required by qpy to
             deserialize the payload. For this reason, the option defaults to False.
+        version: The QPY format version to emit. By default this defaults to
+            the latest supported format (which is currently 10), however for
+            compatibility reasons if you need to load the generated QPY payload with an older version
+            of Qiskit you can also select the minimum supported export version, which only can change
+            during a Qiskit major version release, to generate an older QPY format version. As of this
+            major release series the minimum supported export version is version 10.
```

Comment:
If we changed the default to just be an `int`, we wouldn't need to comment on the "this is currently 10" bit, which I'm almost certain we'll forget to update next time we change it.

We might want to expose the minimum supported export version as a data attribute on the `qiskit.qpy` module (e.g. `qiskit.qpy.MINIMUM_DUMP_VERSION = 10`) so it can be queried programmatically and so we can just point to that in the docs and have that sentence autoupdate as well. (tbh, we probably want `qiskit.qpy.CURRENT_VERSION` as well)

Reply:
`qpy.common.QPY_VERSION` is the equivalent of what `qiskit.qpy.CURRENT_VERSION` would be. But we can rexport it and add a module attribute easily enough.

Reply:
Done in: https://github.com/Qiskit/qiskit/pull/11644/commits/2b90f671d49e1407065654c7b1d77568cb8617d3

Reply:
Now `version=10` shows up in the signature in the API docs, but I think it would be good to add `QPY_VERSION` and `QPY_COMPATIBILITY_VERSION` to the API docs as well, and possibly reference them instead of writing 10 in the docstring here.

Reply:
I added it to the api docs here: https://github.com/Qiskit/qiskit/pull/11644/commits/2d38739507f0d657b1251029014cd54bd0d2281e and removed the use of `10` in the docstring here in: https://github.com/Qiskit/qiskit/pull/11644/commits/600bb9cd77ed37b9d78944ceb065dfbfae632a5a